README.md

```diff
@@ -14,36 +14,167 @@ The term "tenant cluster" refers to the k8s cluster installed on kubevirt VMs, a
-- create kubeconfig for service account
-    - Run `./hack/create-infra-kubeconfig.sh > temp_kubeconfig.yaml`  
-    - Test your kubeconfig. Try listing resources of type VMI in the kubevirt cluster namespace.
-    - This kubeconfig is configured with server url accessible from local machine `127.0.0.1:[0-9]+`, to use it in tenant 
-  cluster the url has to changed to that of a VM: `sed -i -r 's/127.0.0.1:[0-9]+/192.168.66.101:6443/g' temp_kubeconfig.yaml`
-
-- create namespace for the driver in tenant cluster
-    - Use `deploy/000-namespace.yaml`
-- use `deploy/secret.yaml` for creating the necessary secret in the tenant cluster
-    - set kubeconfig: [base64 of kubeconfig from previous step]
-- use `deploy/configmap.yaml` for creating the driver's config
-    - set infraClusterNamespace to the kubevirt cluster namespace.
-    - set infraClusterLabels. The format is 'key=value,key=value,...'. The driver creates resources in the infra cluster. These resources are labeled with the values you supply in infraClusterLabels. Provide values that make the labels unique to your cluster. One usage of such labels is for destroying the tenant cluster. The labels tells us what resources were created in the infra cluster for serving the tenant.
-- deploy files under `deploy` in  tenant cluster
-    - 000-csi-driver.yaml
-    - 020-authorization.yaml
-    - 030-node.yaml
-    - 040-controller.yaml
-- create StorageClass and PersistentVolumeClaim - see `deploy/example`
-- Enable HotplugVolumes feature gate
-    - In case your Kubevirt namespace has the ConfigMap 'kubevirt-config' then use `deploy/example/kubevirt-config.yaml` for adding the feature gate to it. Look at the path {.data.feature-gates}
-    - Otherwise, add the feature gate to the resource of type Kubevirt. There should be a single resource of this type and its name is irrelevant. See `deploy/example/kubevirt.yaml`
-    - Pay attention that in some deployments there are operators that will restore previous configuration. You will have to stop these operators for editing the resources.Some operators allow configuration through their own CRD. HCO is such an operator. See [HCO cluster configuration](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/docs/cluster-configuration.md) to understand how HCO feature gates are configured.
-
-## Examples
+For this example it is assumed that the namespace in which the tenant VMs are deployed is called `kvcluster`.
+
+For split deployment
+```bash
+# Deploy infra service account
+kubectl -n kvcluster apply -f ./deploy/infra-cluster-service-account.yaml
+# Deploy the tenant resources not including the controller with the overlay in deploy/tenant/overlay
+kubectl_tenant apply --kubeconfig $TENANT_KUBECONFIG --kustomize ./deploy/tenant/overlay
+# Deploy the controller resources in the infra cluster with the overlay in deploy/controller-infra/overlay
+kubectl apply --kustomize ./deploy/controller-infra/overlay
+```
```

Comment:
can this be extended to include everything that is required to setup the csi driver so that PVCs can be provisioned in the guest?

Right now it appears like this only deploys the components, but does not configure them. I see you have more information about how to setup the storageclasses, configmaps, etc... later on, but i'm unsure how easy this is for someone to follow who is unfamiliar with all the concepts.

How can we reduce all of this to copying and pasting some commands where the user just needs to supply the bare minimum related to their infra cluster... like the infra storage class and perhaps the namespace VMs run in.

Reply:
So the thing is, if your overlay is set up, this is 3 commands to run.
1. The infra cluster SA RBAC
2. The tenant resources
3. The infra resources
After this you can start creating PVCs in the tenant cluster. Since we have 2 options
1. Everything in tenant (original)
2. Split between tenant and infra (current)
I had to explain the two options. I then below go into the details of what needs to be provided in the overlay for this all to work. I guess I did not make this clear.